sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -102,73 +138,231 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
+    offeringIntensity: OfferingIntensity,
+    studentId: number,
+    auditUserId: number,
+    applicationId: number,
+  ): Promise<StudentRestriction> {
+    if (
+      offeringIntensity === OfferingIntensity.fullTime &&
+      scholasticStandingData.scholasticStanding ===
+        SCHOLASTIC_STANDING_STUDENT_DID_NOT_COMPLETE_PROGRAM &&
+      !!scholasticStandingData.numberOfUnsuccessfulWeeks
+    ) {
+      const totalExistingUnsuccessfulWeeks =
+        await this.totalExistingFTUnsuccessfulWeeks(studentId);
+
+      // When total number of unsuccessful weeks hits minimum 68, add SSR restriction.
+      if (
+        +(totalExistingUnsuccessfulWeeks ?? 0) +
+          +(scholasticStandingData.numberOfUnsuccessfulWeeks ?? 0) >=
+        MINIMUM_UNSUCCESSFUL_WEEKS
+      ) {
+        return this.studentRestrictionService.saveNewStudentRestriction(
+          studentId,
+          RestrictionCode.SSR,
+          auditUserId,
+          applicationId,
+        );
+      }
+    } else if (
+      offeringIntensity === OfferingIntensity.fullTime &&
+      scholasticStandingData.scholasticStanding ===
+        SCHOLASTIC_STANDING_STUDENT_WITHDREW_FROM_PROGRAM &&
+      !!scholasticStandingData.dateOfWithdrawal
+    ) {
+      // Check if "WTHD" restriction is already present for the student,
+      // if not add "WTHD" restriction else add "SSR" restriction.
+      const isWTHDAlreadyExists =
+        await this.studentRestrictionService.isRestrictionExistsForStudent(
+          studentId,
+          RestrictionCode.WTHD,
+        );
+      if (isWTHDAlreadyExists) {
+        return this.studentRestrictionService.saveNewStudentRestriction(
+          studentId,
+          RestrictionCode.SSR,
+          auditUserId,
+          applicationId,
+        );
+      }
+      return this.studentRestrictionService.saveNewStudentRestriction(
+        studentId,
+        RestrictionCode.WTHD,
+        auditUserId,
+        applicationId,
+      );
+    } else if (
+      offeringIntensity === OfferingIntensity.partTime &&
+      ((scholasticStandingData.scholasticStanding ===
+        SCHOLASTIC_STANDING_STUDENT_DID_NOT_COMPLETE_PROGRAM &&
+        !!scholasticStandingData.numberOfUnsuccessfulWeeks) ||
+        (scholasticStandingData.scholasticStanding ===
+          SCHOLASTIC_STANDING_STUDENT_WITHDREW_FROM_PROGRAM &&
+          !!scholasticStandingData.dateOfWithdrawal))
+    ) {
+      return this.studentRestrictionService.saveNewStudentRestriction(
+        studentId,
+        RestrictionCode.PTSSR,
+        auditUserId,
+        applicationId,
+      );
+    }
+  }
+
+  /**
+   * The service gets sum of unsuccessfulWeeks for all existing scholastic standing for the
+   * requested student.
+   * @param studentId student id.
+   * @returns sum of unsuccessfulWeeks for all existing scholastic standing for the
+   * requested student.
+   */
+  async totalExistingFTUnsuccessfulWeeks(studentId: number): Promise<number> {
```

Comment:
I would recommend removing the FT from the name because it does not represent what the method is actually doing. From a business perspective maybe makes sense, but this method is not filtering or doing anything Fulltime specific, IMO.